src/libstd/dynamic_lib.rs

```diff
@@ -263,6 +266,16 @@ mod dl {
                  symbol: *const libc::c_char) -> *mut libc::c_void;
         fn dlclose(handle: *mut libc::c_void) -> libc::c_int;
     }
+
+    #[cfg(target_os = "openbsd")]
```

Comment:
ok, I saw why I can remove `#[link_name="dl"]` and the second block: the linkage with `libdl` will be done (for Linux and others) via `rtdeps.rs`